pkg/sync/filepath_sync.go

```diff
@@ -50,6 +50,13 @@ func (fs *FilePathSync) Notify(w chan<- INotify) {
 				case fsnotify.Write:
 					evtType = E_EVENT_TYPE_MODIFY
 				case fsnotify.Remove:
+					// K8s exposes config maps as symlinks.
+					// Updates cause a remove event, we need to re-add the watcher in this case.
+					err = watcher.Add(fs.URI)
+					if (err != nil) {
+						log.Println("Error restoring watcher: %s", err.Error())
+						log.Fatal(err)
```

Comment:
I guess it makes sense to consider this fatal? If we have an error here, how can we recover? No events will fire subsequently.

Reply:
I'm thinking if for some reason there's an error here (shouldn't be) it would be better to exit the process, killing the pod, instead of having an fs notify that's not getting events.